pallets/ddc-validator/src/lib.rs

```diff
@@ -86,6 +88,26 @@ pub const DEFAULT_DATA_PROVIDER_URL: &str = "http://webdis:7379";
 pub const DATA_PROVIDER_URL_KEY: &[u8; 32] = b"ddc-validator::data-provider-url";
 pub const QUORUM_SIZE: usize = 1;
 
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub enum LogType {
+	Read = 1,
+	Write = 2,
+	Query = 3,
+}
+
+impl TryFrom<u64> for LogType {
```

Comment:
`FromStr` could be a more explicit choice.

https://paritytech.github.io/polkadot-sdk/master/sp_std/str/trait.FromStr.html